integration/test_general_cli.py

```diff
@@ -44,3 +44,26 @@ def test_web_port_required(request, reactor, temp_dir, introducer_furl):
     assert False, "Expected an error from magic-folder"
     print("output: '{}'".format(proto.output.getvalue()))
 
+
+@pytest_twisted.inlineCallbacks
+def test_daemon_inititialize(request, reactor, temp_dir):
+    """
+    'magic-folder init' happy-path works
+    """
+
+    node_dir = join(temp_dir, "daemon")
+
+    proto = util._CollectOutputProtocol()
+    proc = util._magic_folder_runner(
+        proto, reactor, request,
+        [
+            "init",
+            "--config", node_dir,
+            "--listen-endpoint", "tcp:1234",
+            "--tahoe-url", "http://localhost:3456",
+        ],
+    )
+    yield proto.done
+
+    assert exists(join(node_dir, "global.sqlite"))
+    assert exists(join(node_dir, "api_token"))
```

Comment:
If this PR also adds some `magic-folder show-config` command that dumps the config (json?) then the test can assert that the output config matches the input config and remain a bit more oblivious to the exact internal representation.

Reply:
Okay, good points.  `twist` has been a struggle all the way and if it's not even giving us what we want (eg ability to spit out clean json) maybe we should stop fighting with it and use something else.  Not right now, though, I suppose.  So perhaps we can just have a ticket to capture this and deal with it later.